src/Features/Core/Portable/GenerateOverrides/GenerateOverridesWithDialogCodeAction.cs

```diff
@@ -0,0 +1,95 @@
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.PickMembers;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.GenerateOverrides
+{
+    internal partial class GenerateOverridesCodeRefactoringProvider
+    {
+        private class GenerateOverridesWithDialogCodeAction : CodeActionWithOptions
+        {
+            private readonly GenerateOverridesCodeRefactoringProvider _service;
+            private readonly Document _document;
+            private readonly INamedTypeSymbol _containingType;
+            private readonly ImmutableArray<ISymbol> _viableMembers;
+            private readonly TextSpan _textSpan;
+
+            public GenerateOverridesWithDialogCodeAction(
+                GenerateOverridesCodeRefactoringProvider service,
+                Document document,
+                TextSpan textSpan,
+                INamedTypeSymbol containingType,
+                ImmutableArray<ISymbol> viableMembers)
+            {
+                _service = service;
+                _document = document;
+                _containingType = containingType;
+                _viableMembers = viableMembers;
+                _textSpan = textSpan;
+            }
+
+            public override object GetOptions(CancellationToken cancellationToken)
+            {
+                var service = _service._pickMembersService_forTestingPurposes ?? _document.Project.Solution.Workspace.Services.GetService<IPickMembersService>();
+                return service.PickMembers(FeaturesResources.Pick_members_to_override, _viableMembers);
+            }
+
+            protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(object options, CancellationToken cancellationToken)
+            {
+                var result = (PickMembersResult)options;
+                if (result.IsCanceled || result.Members.Length == 0)
+                {
+                    return ImmutableArray<CodeActionOperation>.Empty;
+                }
+
+                var syntaxTree = await _document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+
+                // If the user has selected just one member then we will insert it at the current
+                // location.  Otherwise, if it's many members, then we'll auto insert them as appropriate.
+                var afterThisLocation = result.Members.Length == 1
+                    ? syntaxTree.GetLocation(_textSpan)
+                    : null;
+
+                var generator = SyntaxGenerator.GetGenerator(_document);
+                var memberTasks = result.Members.SelectAsArray(
+                    m => GenerateOverride(generator, m, cancellationToken));
+
+                var members = await Task.WhenAll(memberTasks).ConfigureAwait(false);
```

Comment:
Technically we have to go get a Compilation (so that we can 'accurately' generate things like the "throw" statement inside the method body).  Getting a compilation is async, so this code becomes async.

Or did you mean, the fact that we do it in parallel?  I just did it htat way because it was easy to write :)